sources/packages/api/src/services/student-assessment/student-assessment.service.ts

```diff
@@ -0,0 +1,237 @@
+import { Injectable } from "@nestjs/common";
+import { RecordDataModelService } from "../../database/data.model.service";
+import {
+  ApplicationStatus,
+  AssessmentStatus,
+  AssessmentTriggerType,
+  EducationProgram,
+  EducationProgramOffering,
+  InstitutionLocation,
+  ProgramInfoStatus,
+  StudentAssessment,
+} from "../../database/entities";
+import { Connection, IsNull, UpdateResult } from "typeorm";
+import { CustomNamedError } from "../../utilities";
+import { INVALID_OPERATION_IN_THE_CURRENT_STATUS } from "../application/application.service";
+import { WorkflowActionsService } from "..";
+import { ASSESSMENT_NOT_FOUND } from "./student-assessment.constants";
+
+/**
+ * Manages the student assessment related operations.
+ */
+@Injectable()
+export class StudentAssessmentService extends RecordDataModelService<StudentAssessment> {
+  constructor(
+    connection: Connection,
+    private readonly workflow: WorkflowActionsService,
+  ) {
+    super(connection.getRepository(StudentAssessment));
+  }
+
+  /**
+   * Get the assessment and the related application information.
+   * * This method is used by the the assessment workflow as a main source
+   * * of information for the assessment/reassessment and the application.
+   * @param assessmentId assessment id .
+   * @returns assessment and the related application information.
+   */
+  async getById(assessmentId: number): Promise<StudentAssessment> {
+    return this.repo
+      .createQueryBuilder("assessment")
+      .select([
+        "assessment.id",
+        "application.data",
+        "programYear.programYear",
+        "programYear.startDate",
+        "programYear.endDate",
+        "offering.id",
+        "offering.studyStartDate",
+        "offering.studyEndDate",
+        "offering.actualTuitionCosts",
+        "offering.programRelatedCosts",
+        "offering.mandatoryFees",
+        "offering.exceptionalExpenses",
+        "offering.tuitionRemittanceRequestedAmount",
+        "offering.offeringDelivered",
+        "offering.offeringIntensity",
+        "educationProgram.id",
+        "educationProgram.credentialType",
+        "educationProgram.completionYears",
+        "institution.id",
+        "institutionType.name",
+        "location.data",
+        "student.id",
+        "student.studentPDVerified",
+      ])
+      .innerJoin("assessment.application", "application")
+      .innerJoin("application.programYear", "programYear")
+      .innerJoin("application.student", "student")
+      .leftJoin("assessment.offering", "offering")
+      .leftJoin("offering.educationProgram", "educationProgram")
+      .leftJoin("application.location", "location")
+      .leftJoin("location.institution", "institution")
+      .leftJoin("institution.institutionType", "institutionType")
```

Comment:
I am not sure if I am following but, the inner join "scope' is not about the immediate previous join, right?
If we changed the institution type to be INNER JOIN and there is no location id present, this will make that the entire row be not returned. Does it make sense?

Reply:
Yes you are right. If the leftjoin in the previous joins does not return any rows the institutionType will not fetch any rows, so if the usecase is to return even if the rows are not returned, then we are good.